packages/js-sdk/src/http/request.ts

```diff
@@ -208,6 +209,20 @@ export class KinveyHttpRequest extends HttpRequest {
                 logger.error(error.message);
               }
             }
+          } else {
```

Comment:
I think we should remove this section as it is only executed in case the login is not with MIC.
This means that the behavior will differ in the case of MIC and not MIC:
- if you are logged with MIC - the session is not automatically cleared
- if you are not logged with MIC - the session is automatically cleared

It would be better if the session is cleared in both cases if the MIC refresh fails for some reason.

Reply:
Yes, I may have misinterpreted the task's description but it makes sense to clear it in both cases.